lib/apiservers/service/restapi/handlers/vch_get.go

```diff
@@ -0,0 +1,428 @@
+					RoutingDestinations: []models.IPRange{asIPRange(&value.Gateway)},
+				},
+				IPRanges: *asIPRanges(&value.Destinations),
+			})
+		}
+	}
+	model.Network.Container = containerNetworks
+
+	// storage
+	scratchSize := int(vchConfig.Storage.ScratchSize)
+	model.Storage = &models.VCHStorage{
+		BaseImageSize: asKiB(&scratchSize),
+	}
+
+	volumeLocations := make([]*models.VCHStorageVolumeStoresItems0, 0, len(vchConfig.Storage.VolumeLocations))
+	for label, path := range vchConfig.Storage.VolumeLocations {
+		volume := models.VCHStorageVolumeStoresItems0{Datastore: path.String(), Label: label}
+		volumeLocations = append(volumeLocations, &volume)
+	}
+	model.Storage.VolumeStores = volumeLocations
+
+	imageStores := make([]string, 0, len(vchConfig.Storage.ImageStores))
+	for _, value := range vchConfig.Storage.ImageStores {
+		imageStores = append(imageStores, value.String())
+	}
+	model.Storage.ImageStores = imageStores
+
+	// auth
+	model.Auth = &models.VCHAuth{
+		Client: &models.VCHAuthClient{},
+	}
+
+	if vchConfig.Certificate.HostCertificate != nil {
+		model.Auth.Server = &models.VCHAuthServer{
+			Certificate: asPemCertificate(vchConfig.Certificate.HostCertificate.Cert),
+		}
+	}
+
+	model.Auth.Client.CertificateAuthorities = asPemCertificates(vchConfig.Certificate.CertificateAuthorities)
+
+	// endpoint
+	model.Endpoint = &models.VCHEndpoint{
+		UseResourcePool: d.UseRP,
+		Memory:          asMiB(&d.MemoryMB),
+		CPU: &models.VCHEndpointCPU{
+			Sockets: int64(d.NumCPUs),
+		},
+		OperationsCredentials: &models.VCHEndpointOperationsCredentials{
+			User: vchConfig.Connection.Username,
+			// Password intentionally excluded from GET responses for security reasons!
+		},
+	}
+
+	// registry
+	model.Registry = &models.VCHRegistry{
+		Insecure:               vchConfig.Registry.InsecureRegistries,
+		Whitelist:              vchConfig.Registry.RegistryWhitelist,
+		Blacklist:              vchConfig.Registry.RegistryBlacklist,
+		CertificateAuthorities: asPemCertificates(vchConfig.Certificate.RegistryCertificateAuthorities),
+		ImageFetchProxy:        asImageFetchProxy(vchConfig.ExecutorConfig.Sessions[config.VicAdminService], config.VICAdminHTTPProxy, config.VICAdminHTTPSProxy),
+	}
+
+	// runtime
+	model.Runtime = &models.VCHRuntime{}
+
+	installerVer := version.GetBuild()
+	upgradeStatus := upgradeStatusMessage(context.Background(), vch, installerVer, vchConfig.Version)
+	model.Runtime.UpgradeStatus = upgradeStatus
+
+	powerState, err := vch.PowerState(context.Background())
+	if err != nil {
+		powerState = "error"
+	}
+	model.Runtime.PowerState = string(powerState)
+
+	if public := vchConfig.ExecutorConfig.Networks["public"]; public != nil {
+		if public_ip := public.Assigned.IP; public_ip != nil {
+			var docker_port string
+			if !vchConfig.HostCertificate.IsNil() {
+				docker_port = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
+			} else {
+				docker_port = fmt.Sprintf("%d", opts.DefaultHTTPPort)
```

Comment:
s/docker_port/dockerPort

(golint will complain about using underscores in var names)